front/lib/api/assistant/actions/database_query.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  AgentConfigurationType,
+  AgentMessageType,
+  cloneBaseConfig,
+  ConversationType,
+  DatabaseQueryActionType,
+  DatabaseQueryRunErrorEvent,
+  DatabaseQueryRunSuccessEvent,
+  DustProdActionRegistry,
+  Err,
+  isDatabaseQueryConfiguration,
+  ModelMessageType,
+  Ok,
+  Result,
+  UserMessageType,
+} from "@dust-tt/types";
+
+import { runActionStreamed } from "@app/lib/actions/server";
+import { generateActionInputs } from "@app/lib/api/assistant/agent";
+import { Authenticator } from "@app/lib/auth";
+import { AgentDatabaseQueryAction } from "@app/lib/models";
+
+/**
+ * Model rendering of DatabaseQueryAction.
+ */
+
+export function renderDatabaseQueryActionForModel(
+  action: DatabaseQueryActionType
+): ModelMessageType {
+  let content = "";
+  if (!action.output) {
+    throw new Error(
+      "Output not set on DatabaseQuery action; execution is likely not finished."
+    );
+  }
+  content += `OUTPUT:\n`;
+  content += `${JSON.stringify(action.output, null, 2)}\n`;
+
+  return {
+    role: "action" as const,
+    name: "DatabaseQuery",
+    content,
+  };
+}
+
+/**
+ * Generate the parameters for the DatabaseQuery app.
+ */
+export async function generateDatabaseQueryAppParams(
+  auth: Authenticator,
+  configuration: AgentConfigurationType,
+  conversation: ConversationType,
+  userMessage: UserMessageType
+): Promise<
+  Result<
+    {
+      [key: string]: string | number | boolean;
+    },
+    Error
+  >
+> {
+  const c = configuration.action;
+  if (!isDatabaseQueryConfiguration(c)) {
+    throw new Error(
+      "Unexpected action configuration received in `runQueryDatabase`"
+    );
+  }
+
+  const spec = {
+    name: "query_database",
+    description:
+      "The user's request. This is what will be given to the assistant responsible for generating the SQL query, It must include all relevant information.",
```

Comment:
nit: In the retrieval spec, this description is more a description of the action itself than a description of the param to generate. I wonder if it's not a little redundant with the description of the "question" input below ?
Def not a strong opinion and we can also iterate on that later

Reply:
Yes it should be descriptive of the action. `The user's request` is not intelligible here and will probably reduce quality of answers of the model.

Reply:
Updated, let me know if you think this is good enough for a first ship. I plan to iterate on it while digging the use cases we identified with some design partners.